src/queries/CategoryQuery.ts

```diff
@@ -0,0 +1,95 @@
+import Category from '../models/CategoryModel';
+import CategoryInterface from 'interfaces/CategoryInterface';
+import { Op } from 'sequelize';
+import { sequelize } from '../db/connection';
+import Product from '../models/ProductModel';
+export default class CategoryQuery {
+  static create = async (name: string) => {
+    return Category.create(
+      { name },
+      {
+        returning: true
+      }
+    );
+  };
+  static update = async (category: CategoryInterface) => {
+    const { id, name } = category;
+    return Category.update(
+      { name },
+      {
+        where: {
+          id
+        },
+        returning: true
+      }
+    );
+  };
+
+  static delete = async (id: string) => {
+    return Category.destroy({
+      where: {
+        id
+      }
+    });
+  };
+
+  static getCount = async () => {
+    return Category.count();
+  };
+
+  static getAll = async (limit: number, offset: number) => {
+    return Category.findAll({
+      attributes: [
+        'name',
+        'Category.id',
+        'Category.createdAt',
+        [
+          sequelize.fn('count', sequelize.col('Products.categoryId')),
+          'productsCount'
+        ]
+      ],
+      // order: [['Category.id', 'DESC']],
+      // order: [['Category.createdAt', 'DESC']],
```

Comment:
remove commented code, it make things confusing in the future when the code base grows

Reply:
Ordering here didn't work with me, do you have idea why??